File: WebApp/indoor-localization/src/entities/AssetZoneHistoryLogFilter.ts
export class AssetZoneHistoryLogFilter {
  zoneId: number | null;
  assetId: number | null;
  enterStartDate: Date | null;
  enterEndDate: Date | null;
  exitStartDate: Date | null;
  exitEndDate: Date | null;
  minRetentionTime: number | null;
  maxRetentionTime: number | null;

  constructor(
    zoneId: number | null,
    assetId: number | null,
    enterStartDate: Date | null,
    enterEndDate: Date | null,
    exitStartDate: Date | null,
    exitEndDate: Date | null,
    minRetentionTime: number | null,
    maxRetentionTime: number | null
  ) {
    this.zoneId = zoneId;
    this.assetId = assetId;
    this.enterStartDate = enterStartDate;
    this.enterEndDate = enterEndDate;
    this.exitStartDate = exitStartDate;
    this.exitEndDate = exitEndDate;
    this.minRetentionTime = minRetentionTime;
    this.maxRetentionTime = maxRetentionTime;
  }
}

File: Android/indoor_localization/lib/domain/entities/asset_zone_history_log_filter.dart
class AssetZoneHistoryLogFilter {
  final int zoneId;
  final int assetId;
  final DateTime enterStartDate;
  final DateTime enterEndDate;
  final DateTime exitStartDate;
  final DateTime exitEndDate;
  final int minRetentionTime;
  final int maxRetentionTime;

  AssetZoneHistoryLogFilter({
    required this.zoneId,
    required this.assetId,
    required this.enterStartDate,
    required this.enterEndDate,
    required this.exitStartDate,
    required this.exitEndDate,
    required this.minRetentionTime,
    required this.maxRetentionTime,
  });
}
